### PR TITLE
HelmReleases Cleanup: Rework & suspend.

### PR DESCRIPTION
**Pull request by Gacko**
Towards https://github.com/giantswarm/roadmap/issues/3435.

To simplify review, I made commits as granular as possible. So just review them one by one instead of the whole big change.

Overall the job should no longer remove finalizers but rather suspend the HelmRelease. It's yet unclear if the HelmChart also needs to be suspended or if Flux cleans that up.

See https://github.com/giantswarm/cluster-aws/pull/680 for tests.

### CODE REVIEW

**Comment by Gacko** [comment]
CAPA China tests are stuck due to missing Kubernetes v1.26.15 AMI. Others are passing.

File: helm/cluster/templates/apps/helmreleases-cleanup/job.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
+{{- if .Values.providerIntegration.resourcesApi.cleanupHelmReleaseResourcesEnabled }}
+#
+# Because cluster resources are often deleted before Flux has a chance to uninstall the Helm releases for all deleted HelmRelease CRs,
```
Comment (ericgraf) [inline helm/cluster/templates/apps/helmreleases-cleanup/job.yaml:3]: I don't think this comment block makes sense anymore.

**Review by ericgraf — APPROVED** [review]

**Comment by Gacko** [comment]
CAPA China is green too now thanks to @ericgraf!

**Comment by Gacko** [comment]
@giantswarm/team-turtles Anyone up for a review?

**Pull request merged**